File: DropZone/DropZone.Web/src/models/hashBlocks/hashBlock.model.ts
import { UserModel } from "../users/user.model";

export class HashBlockModel {
	public Id: number;
	public Value: string;
	public Hash: string;
	public PreviousHash: string;
	public CreatedOn: Date;
    public ModifiedAt: Date;

	public UserId: number;
    public PreviousHashBlockId: number;
    
	public User: UserModel;
    public PreviousHashBlock: HashBlockModel;
};

File: DropZone/DropZone.Core/Models/HashBlock/HashBlockModel.cs
using System;

namespace DropZone.Core.Models
{
    public class HashBlockModel
    {
        public long Id { get; set; }

        public string Value { get; set; }

        public string Hash { get; set; }

        public string PreviousHash { get; set; }

        public DateTime CreatedOn { get; set; }

        public DateTime ModifiedAt { get; set; }


        public long UserId { get; set; }

        public long? PreviousHashBlockId { get; set; }


        public UserModel User { get; set; }

        public HashBlockModel PreviousHashBlock { get; set; }
    }
}
